packages/eslint-plugin/src/rules/strict-void-return.ts

```diff
@@ -0,0 +1,991 @@
+import assert from 'node:assert';
```

Comment:
[Refactor] Rules can run in environments other than Node. But, even if we did want to take a dependency on `node:assert`, the kinds of assertions in this rule aren't what we'd do. Lint rules run in a controlled enough environment (known AST inputs) that we should be able to handle all cases without crashing. 

If something is known to be a specific type and there's no good way to tell the type system that, use an `as`.

Aside: it's common that the type system is actually correct when it's yelling about a node being the wrong type. Or at least - it's correct per the AST's types. If it's yelling and you think it shouldn't then that might be a bug in the AST types? For example, in the `if (funcNode.generator)` check, `funcNode.body` is typed as `TSESTree.BlockStatement | TSESTree.Expression`... even though if `funcNode.generator` is true, we know it's not an arrow expression. A bug in the AST types maybe?

Reply:
Why not add `assert` to utils like `nullThrows`?